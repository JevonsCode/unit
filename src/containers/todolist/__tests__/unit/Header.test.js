import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/header';
import { findTestWrapper } from "../../../../utils/testUtils";

describe("Header 组件", () => {

    const wrapper = shallow(<Header />);

    it("渲染样式快照", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('包含一个 input', () => {
        const inputElem = findTestWrapper(wrapper, "input");
        expect(inputElem.length).toBe(1);
    });

    it("input 初始化应该为空", () => {
        const wrapper = shallow(<Header />);
        const inputElem = findTestWrapper(wrapper, "input");
        expect(inputElem.prop("value")).toEqual("");
    });

    it("input 当用户输入时，响应变化", () => {
        const wrapper = shallow(<Header />);
        const inputElem = findTestWrapper(wrapper, "input");
        const userInput = "enzyme!";
        inputElem.simulate("change", {
            target: { value: userInput }
        });
        expect(wrapper.state("value")).toEqual(userInput);
        // 测试 DOM 展示
        // const newInputElem = wrapper.find("[data-test='input']");
        // expect(newInputElem.prop("value")).toEqual(userInput);
    });

    it("input 输入回车时 没有内容 无操作", () => {
        const fn = jest.fn();
        const wrapper = shallow(<Header addUndoItem={fn} />);
        const inputElem = findTestWrapper(wrapper, "input");
        wrapper.setState({
            value: ""
        });
        inputElem.simulate("keyUp", {
            keyCode: 13
        });
        expect(fn).not.toHaveBeenCalled();
    });

    it("input 输入回车时 有内容 函数应该被调用 函数清空", () => {
        const fn = jest.fn();
        const wrapper = shallow(<Header addUndoItem={fn} />);
        const inputElem = findTestWrapper(wrapper, "input");
        wrapper.setState({
            value: "1"
        });
        inputElem.simulate("keyUp", {
            keyCode: 13
        });
        expect(fn).toHaveBeenCalled();
        expect(fn).toHaveBeenCalledWith("1");
        const inputElem2 = findTestWrapper(wrapper, "input");
        // 如果有内容最后清除掉
        expect(inputElem2.prop("value")).toBe("");
    });
});

