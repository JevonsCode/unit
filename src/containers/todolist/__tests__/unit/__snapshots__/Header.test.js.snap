// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件 渲染样式快照 1`] = `
<div
  className="header"
>
  <div
    className="header-box"
  >
    TODOLIST
    <input
      className="header-input"
      data-test="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Todo"
      value=""
    />
  </div>
</div>
`;
